File: app/src/main/java/com/galeria/android/HiddenActivity.java
package com.galeria.android;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.FrameLayout;
import android.widget.GridView;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.io.File;

public class HiddenActivity extends Activity {
    private HiddenFileAdapter adapter;
    private TextView emptyView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        buildLayout();
    }

    @Override
    protected void onResume() {
        super.onResume();
        loadFiles();
    }

    private void buildLayout() {
        LinearLayout root = new LinearLayout(this);
        root.setOrientation(LinearLayout.VERTICAL);
        root.setBackgroundColor(Ui.BG);

        LinearLayout bar = new LinearLayout(this);
        bar.setGravity(Gravity.CENTER_VERTICAL);
        Ui.setPadding(bar, 10, 8, 10, 8);

        TextView back = Ui.title(this, "Voltar", 16);
        back.setGravity(Gravity.CENTER);
        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });
        bar.addView(back, new LinearLayout.LayoutParams(Ui.dp(this, 76), Ui.dp(this, 44)));

        TextView title = Ui.title(this, "Ocultos", 22);
        bar.addView(title, new LinearLayout.LayoutParams(0, ViewGroup.LayoutParams.WRAP_CONTENT, 1));
        root.addView(bar);

        FrameLayout content = new FrameLayout(this);
        GridView grid = new GridView(this);
        grid.setNumColumns(GridView.AUTO_FIT);
        grid.setColumnWidth(Ui.dp(this, 132));
        grid.setStretchMode(GridView.STRETCH_COLUMN_WIDTH);
        grid.setHorizontalSpacing(Ui.dp(this, 2));
        grid.setVerticalSpacing(Ui.dp(this, 2));
        grid.setPadding(Ui.dp(this, 6), Ui.dp(this, 6), Ui.dp(this, 6), Ui.dp(this, 16));
        grid.setClipToPadding(false);
        adapter = new HiddenFileAdapter(this);
        grid.setAdapter(adapter);
        grid.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                File file = adapter.getItem(position);
                Intent intent = new Intent(HiddenActivity.this, FileDetailActivity.class);
                intent.putExtra("path", file.getAbsolutePath());
                startActivity(intent);
            }
        });
        content.addView(grid);

        emptyView = Ui.label(this, "Nenhum item oculto.");
        content.addView(emptyView, new FrameLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT));
        root.addView(content, new LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 0, 1));
        setContentView(root);
    }

    private void loadFiles() {
        File dir = MediaActions.hiddenDir(this);
        File[] files = dir.exists() ? dir.listFiles() : null;
        adapter.submit(files);
        emptyView.setVisibility(adapter.getCount() == 0 ? View.VISIBLE : View.GONE);
    }
}

File: app/src/main/java/com/galeria/android/HiddenActivity.kt
package com.galeria.android

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.Gravity
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.FrameLayout
import android.widget.GridView
import android.widget.LinearLayout
import android.widget.TextView

class HiddenActivity : Activity() {
    private lateinit var adapter: HiddenFileAdapter
    private lateinit var emptyView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        buildLayout()
    }

    override fun onResume() {
        super.onResume()
        loadFiles()
    }

    private fun buildLayout() {
        val root = LinearLayout(this).apply {
            orientation = LinearLayout.VERTICAL
            setBackgroundColor(Ui.BG)
        }

        val bar = LinearLayout(this).apply {
            gravity = Gravity.CENTER_VERTICAL
            Ui.setPadding(this, 10, 8, 10, 8)
        }
        val back = Ui.title(this, "Voltar", 16).apply {
            gravity = Gravity.CENTER
            setOnClickListener { finish() }
        }
        bar.addView(back, LinearLayout.LayoutParams(Ui.dp(this, 76), Ui.dp(this, 44)))

        val title = Ui.title(this, "Ocultos", 22)
        bar.addView(title, LinearLayout.LayoutParams(0, ViewGroup.LayoutParams.WRAP_CONTENT, 1f))
        root.addView(bar)

        val content = FrameLayout(this)
        val grid = GridView(this).apply {
            numColumns = GridView.AUTO_FIT
            columnWidth = Ui.dp(this@HiddenActivity, 132)
            stretchMode = GridView.STRETCH_COLUMN_WIDTH
            horizontalSpacing = Ui.dp(this@HiddenActivity, 2)
            verticalSpacing = Ui.dp(this@HiddenActivity, 2)
            setPadding(
                Ui.dp(this@HiddenActivity, 6),
                Ui.dp(this@HiddenActivity, 6),
                Ui.dp(this@HiddenActivity, 6),
                Ui.dp(this@HiddenActivity, 16)
            )
            clipToPadding = false
        }
        adapter = HiddenFileAdapter(this)
        grid.adapter = adapter
        grid.onItemClickListener = AdapterView.OnItemClickListener { _, _, position, _ ->
            val file = adapter.getItem(position)
            val intent = Intent(this, FileDetailActivity::class.java).apply {
                putExtra("path", file.absolutePath)
            }
            startActivity(intent)
        }
        content.addView(grid)

        emptyView = Ui.label(this, "Nenhum item oculto.")
        content.addView(
            emptyView,
            FrameLayout.LayoutParams(
                ViewGroup.LayoutParams.MATCH_PARENT,
                ViewGroup.LayoutParams.MATCH_PARENT
            )
        )
        root.addView(content, LinearLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, 0, 1f))
        setContentView(root)
    }

    private fun loadFiles() {
        val dir = MediaActions.hiddenDir(this)
        val files = if (dir.exists()) dir.listFiles() else null
        adapter.submit(files)
        emptyView.visibility = if (adapter.count == 0) View.VISIBLE else View.GONE
    }
}
